### ISSUE TITLE
[Search] Stop words in resource name cause Lunr to return zero results

### ISSUE THREAD

**Issue by Niksko** [backstage/backstage#12628]
## Expected Behavior

Searching inside of a techdocs site works as expected

## Actual Behavior

No search results are returned from in-docs search for some techdocs pages, while others work as expected

## Steps to Reproduce

The following will create a sample techdocs directory that exhibits this problem.

```
mkdir search-no-results-reproduction
cd search-no-results-reproduction
cat > mkdocs.yaml <<EOF
site_name: test-site
nav:
  - Home: index.md
plugins:
  - techdocs-core
EOF
mkdir docs
echo "# Home" > docs/index.md
cat > catalog-info.yaml <<EOF
---
apiVersion: backstage.io/v1alpha1
kind: Component
metadata:
  name: 'techdocs-at-mycompany'
  annotations:
    backstage.io/techdocs-ref: dir:.
spec:
  type: 'repository'
  owner: 'testowner'
  system: 'system'
  lifecycle: 'production'
EOF
```

Including this `catalog-info.yaml` into your Backstage instance and using built-in mkdocs publishing will trigger this issue when trying to use the in-docs search. For example, searching for 'home' will return no results.

Removing the `at` from the `metadata.name` field of the `catalog-info.yaml`, and then ensuring that Backstage builds this update and does a re-index will resolve the issue, searching for 'home' will return results as expected.

I've validated that this is an issue with stopwords (and not e.g. an issue with query clause where the term is too short) by using e.g a `metadata.name` of `techdocs-because-mycompany`. This indicates that it's a stopwords issue.

## Context

We noticed that the search on some TechDocs sites seemed to always return zero results.
The common factor we have found is that the names of the catalog entries associated with these sites contains a stopword.
Through debugging, what we found was that when the query was being constructed by the query translator, the Lunr tokenizer was, for example, turning a resource named `"builds-at-company"` in to `["builds", "at", "company"]`, and adding each of these as a required clause in the Lunr query. When Lunr tries to execute this query, since `"at"` is a stopword (I think this is why), there are no results returned, which causes a required query match to be empty, resulting in no results for all queries.

I think the solution is to filter the tokenized filter value for stopwords before adding them as query clauses.
I'm still very new to how search works, so please let me know if this isn't the right solution here.
Happy to submit a PR to try and fix this.

## Your Environment

```text
yarn run v1.22.17
$ /Users/nskoufis/Code/seek-backstage/node_modules/.bin/backstage-cli info
OS:   Darwin 21.5.0 - darwin/x64
node: v16.14.0
yarn: 1.22.17
cli:  0.17.1 (installed)

Dependencies:
  @backstage/app-defaults                          1.0.2
  @backstage/backend-common                        0.13.5
  @backstage/backend-tasks                         0.3.1
  @backstage/catalog-client                        1.0.2
  @backstage/catalog-model                         1.0.2
  @backstage/cli-common                            0.1.9
  @backstage/cli                                   0.17.1
  @backstage/config-loader                         1.1.1
  @backstage/config                                1.0.1
  @backstage/core-app-api                          1.0.2
  @backstage/core-components                       0.9.4
  @backstage/core-plugin-api                       1.0.2
  @backstage/errors                                1.0.0
  @backstage/integration-react                     1.1.0
  @backstage/integration                           1.2.0
  @backstage/plugin-api-docs                       0.8.5
  @backstage/plugin-app-backend                    0.3.32
  @backstage/plugin-auth-backend                   0.14.0
  @backstage/plugin-auth-node                      0.2.1
  @backstage/plugin-catalog-backend-module-aws     0.1.5
  @backstage/plugin-catalog-backend-module-github  0.1.3
  @backstage/plugin-catalog-backend                1.1.2
  @backstage/plugin-catalog-common                 1.0.2
  @backstage/plugin-catalog-graph                  0.2.17
  @backstage/plugin-catalog-import                 0.8.8
  @backstage/plugin-catalog-react                  1.1.0
  @backstage/plugin-catalog                        1.2.0
  @backstage/plugin-github-actions                 0.5.5
  @backstage/plugin-home                           0.4.21
  @backstage/plugin-org                            0.5.5
  @backstage/plugin-permission-common              0.6.1
  @backstage/plugin-permission-node                0.6.1
  @backstage/plugin-permission-react               0.4.1
  @backstage/plugin-proxy-backend                  0.2.26
  @backstage/plugin-scaffolder-backend             1.2.0
  @backstage/plugin-scaffolder-common              1.1.0
  @backstage/plugin-scaffolder                     1.2.0
  @backstage/plugin-search-backend-node            0.6.1
  @backstage/plugin-search-backend                 0.5.2
  @backstage/plugin-search-common                  0.3.4
  @backstage/plugin-search-react                   0.2.0
  @backstage/plugin-search                         0.8.1
  @backstage/plugin-stack-overflow                 0.1.1
  @backstage/plugin-tech-radar                     0.5.12
  @backstage/plugin-techdocs-backend               1.1.1
  @backstage/plugin-techdocs-module-addons-contrib 1.0.0
  @backstage/plugin-techdocs-node                  1.1.1
  @backstage/plugin-techdocs-react                 1.0.0
  @backstage/plugin-techdocs                       1.1.1
  @backstage/plugin-user-settings                  0.4.4
  @backstage/release-manifests                     0.0.3
  @backstage/search-common                         0.3.4
  @backstage/test-utils                            1.1.0
  @backstage/theme                                 0.2.15
  @backstage/types                                 1.0.0
  @backstage/version-bridge                        1.0.1
✨  Done in 1.46s.
```

**Comment by swenhan** [comment]
PR for this issue raised in: https://github.com/backstage/backstage/pull/13297

**Comment by Niksko** [comment]
Fixed in #13297
